fuel-merkle/src/binary/merkle_tree.rs

```diff
@@ -23,10 +23,11 @@ use crate::{
 
 use alloc::vec::Vec;
 use core::marker::PhantomData;
+use fuel_storage::StorageError;
 
-#[derive(Debug, Clone)]
+#[derive(Debug)]
 #[cfg_attr(feature = "std", derive(thiserror::Error))]
-pub enum MerkleTreeError<StorageError> {
```

Comment:
Agreed. I was a bit overeager to un-genericize the types since I needed to do changes in `fuel-vm` removing some abstraction. It was unnecessary on `fuel-merkle`. I've changed the approach, see https://github.com/FuelLabs/fuel-vm/pull/582#issuecomment-1728570111